src/debezium-service/templates/deployment.yaml

```diff
@@ -0,0 +1,167 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Values.name}}
+  labels:
+    app: {{.Values.name}}
+  namespace: {{.Values.namespace.name}}
+spec:
+  progressDeadlineSeconds: 300
+  replicas: {{int .Values.instances}}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: {{.Values.name}}
+  template:
+    metadata:
+      labels:
+        app: {{.Values.name}}
+        harness-team: PIE
+        harness-module: Platform
+      annotations:
+        cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
+        prometheus.io/scrape: 'true'
+        prometheus.io/port: '8889'
+        prometheus.io/path: '/metrics'
+      {{- if .Values.istioInjectionEnabled}}
+      annotations:
+        sidecar.istio.io/inject: "true"
+      {{- end}}
+    spec:
+      terminationGracePeriodSeconds: 30
+      {{- if .Values.dockercfg}}
+      imagePullSecrets:
+        - name: dockerhub-harness-debezium-svc
+      {{- end}}
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: app
+                    operator: In
+                    values:
+                      - {{.Values.name}}
+              topologyKey: "kubernetes.io/hostname"
+      containers:
+        - name: {{.Values.name}}
+          image: {{.Values.image}}
+          imagePullPolicy: Always
+          lifecycle:
+            postStart:
+              exec:
+                command:
+                  - /bin/sh
+                  - -c
+                  - |
+                    mkdir -p /opt/harness/dumps/${SERVICE_NAME}/${ENV_TYPE}/jfr_dumps/${POD_NAME};
+                    ln -s /opt/harness/dumps/${SERVICE_NAME}/${ENV_TYPE}/jfr_dumps/${POD_NAME} /opt/harness/POD_NAME ;
+                    mkdir -p /opt/harness/dumps/${SERVICE_NAME}/${ENV_TYPE}/crash_dumps/${POD_NAME};
+                    [[ -L /opt/harness/dumps/${SERVICE_NAME}/${ENV_TYPE}/crash_dumps/POD_NAME ]] && unlink /opt/harness/dumps/${SERVICE_NAME}/${ENV_TYPE}/crash_dumps/POD_NAME ;
+                    ln -s /opt/harness/dumps/${SERVICE_NAME}/${ENV_TYPE}/crash_dumps/${POD_NAME} /opt/harness/dumps/${SERVICE_NAME}/${ENV_TYPE}/crash_dumps/POD_NAME ;
+            preStop:
```

Comment:
add it as an optional field, postStart and preStop are not needed in case of SMP. Check Harness-manager chart